File: sdk/web/component/sqljson/controller/IntegrationController.js
var IntegrationController = function IntegrationController(cfg)
{
    let ctrl = {};

    ctrl.Execute = function (integrationName, data, authentication, onResponse)
    {
        let args = {
            Name: integrationName,
            Data: data,
            Authentication: authentication
        };
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Integration/Execute", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Send = function (receiver, data, onResponse)
    {
        let arg = {
            Receiver: receiver,
            Data: data,
        };
        cfg.Connector.Submit(arg, "ardvro/component/sqljson/Integration/Send", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };
    
    ctrl.Broadcast = function (receivers, data, onResponse)
    {
        let arg = {
            Receivers: receivers,
            Data: data,
        };
        cfg.Connector.Submit(arg, "ardvro/component/sqljson/Integration/Stream", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Get = function (integrationId, data, authentication, onResponse)
    {
        let args = {
            Id: integrationId,
            Data: data,
            Authentication: authentication
        };
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Integration/Get", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Post = function (integrationId, data, authentication, onResponse)
    {
        let args = {
            Id: integrationId,
            Data: data,
            Authentication: authentication
        };
        cfg.Connector.Submit(args, "ardvro/component/sqljson/Integration/Post", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    return ctrl;
};

File: sdk/android/component/sqljson/controller/IntegrationController.java
package ardvro.component.sqljson.controller;

import java.util.ArrayList;

import ardvro.core.connection.IConnection;
import ardvro.core.connection.IResponseListener;
import ardvro.core.transceiver.Box;

public class IntegrationController
{
    private IConnection _conection;

    public IntegrationController(IConnection connection)
    {
        _conection = connection;
    }

    public void Execute(String name, Object data, String authorization, IResponseListener onResponse)
    {
        String functionName = "ardvro/component/sqljson/integration/execute";
        _conection.Submit(new Object[]{ name, data, authorization }, functionName, onResponse);
    }

    public void Send(String receiver, Object data, IResponseListener<ArrayList<Box>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/integration/send";
        _conection.Submit(new Object[]{ receiver, data }, functionName, onResponse);
    }

    public void Broadcast(String[] receivers, Object data, IResponseListener<ArrayList<Box>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/integration/broadcast";
        _conection.Submit(new Object[]{ receivers, data }, functionName, onResponse);
    }

    public void Get(Object id, Object data, String authorization, IResponseListener<Box> onResponse)
    {
        String functionName = "ardvro/component/sqljson/integration/Get";
        _conection.Submit(new Object[]{ id, data, authorization }, functionName, onResponse);
    }

    public void Post(Object id, Object data, String authorization, IResponseListener<Box> onResponse)
    {
        String functionName = "ardvro/component/sqljson/integration/Post";
        _conection.Submit(new Object[]{ id, data, authorization }, functionName, onResponse);
    }

}
